gh-123299: Some copyedits to What's New in 3.14 (#133622)
python/cpython: Doc/whatsnew/3.14.rst

.. code-block:: python

   attributes = {"src": "shrubbery.jpg", "alt": "looks nice"}

>    template = t"<img {attributes}>"

Reviewer: In XHTML and HTML5, you should close the `<img>` tag, no? https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/img

You might propose a separated PR to change PEP 750 text.

Reviewer 2: HTML5:

> _Self-closing tags (`<tag />`) do not exist in HTML._

XHMTL:

> Self-closing tags are required in void elements in [XML](https://developer.mozilla.org/en-US/docs/Glossary/XML), [XHTML](https://developer.mozilla.org/en-US/docs/Glossary/XHTML), and [SVG](https://developer.mozilla.org/en-US/docs/Glossary/SVG) (e.g., `<circle cx="50" cy="50" r="50" />`).

(`img` is a void element: it must have a start tag `<img>` but must not have an end tag `</img>`.)

https://developer.mozilla.org/en-US/docs/Glossary/Void_element#self-closing_tags

But this example is HTML and we can ignore XHTML.

Reviewer: Oh ok, I trust you. I didn't write HTML for many years :-)

                     ^^^^^^
   SyntaxError: cannot use subscript as import target

* Improved error message when trying to add an instance of an unhashable type

>   to a :class:`dict` or :class:`set`:

Reviewer 3: ```suggestion
  to a :class:`dict` or :class:`set`.
```
Since the codeblock is later.

>   (Contributed by Victor Stinner in :gh:`426449d`.)

Author: This is the PR number, not the issue number, and I ignore the follow-up that fixed a typo in code

>   (Contributed by Victor Stinner in :gh:`426449d`.)

Reviewer 3: ```suggestion
  (Contributed by Victor Stinner in :gh:`132828`.)
```
Correct pr number

> * Improved error message when trying to add an instance of an unhashable type
>   to a :class:`dict` or :class:`set`:
> 
>   (Contributed by Victor Stinner in :gh:`426449d`.)

Author: ```suggestion
* Improved error message when trying to add an instance of an unhashable type to
  a :class:`dict` or :class:`set`. (Contributed by CF Bolz-Tereick and Victor Stinner
  in :gh:`132828`.)
```

>   (Contributed by Victor Stinner in :gh:`426449d`.)

Reviewer: You should also credit  CF Bolz-Tereick who proposed the change.

  .. code-block:: pycon

     >>> s = set()
     >>> s.add({'pages': 12, 'grade': 'A'})
     Traceback (most recent call last):
       File "<stdin>", line 1, in <module>

>      TypeError: Cannot use 'dict' as a set element (unhashable type).

Reviewer 3: ```suggestion
     TypeError: cannot use 'dict' as a set element (unhashable type: 'dict')
```
Correct msg

>      Traceback (most recent call last):
>        File "<stdin>", line 1, in <module>
>      TypeError: Cannot use 'dict' as a set element (unhashable type).

Reviewer 2: The full traceback?
```suggestion
     Traceback (most recent call last):
       File "<python-input-1>", line 1, in <module>
         s.add({'pages': 12, 'grade': 'A'})
         ~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     TypeError: cannot use 'dict' as a set element (unhashable type: 'dict')
```

     >>> d = {}
     >>> l = [1, 2, 3]
     >>> d[l] = 12
     Traceback (most recent call last):
       File "<stdin>", line 1, in <module>

>      TypeError: Cannot use 'list' as a dict key (unhashable type).

Author: copied straight from the issue, not double-checked

>      TypeError: Cannot use 'list' as a dict key (unhashable type).

Reviewer 3: ```suggestion
     TypeError: cannot use 'list' as a dict key (unhashable type: 'list')
```
Correct error message

>      Traceback (most recent call last):
>        File "<stdin>", line 1, in <module>
>      TypeError: Cannot use 'list' as a dict key (unhashable type).

Reviewer 2: ```suggestion
     Traceback (most recent call last):
       File "<python-input-4>", line 1, in <module>
         d[l] = 12
         ~^^^
     TypeError: cannot use 'list' as a dict key (unhashable type: 'list')
```
